docs/advanced-topics/advanced-methods.md

```diff
@@ -0,0 +1,100 @@
+
+Since only one sample is needed to calculate the loss with the ArcFace and CosFace functions, there are no constraints on what each batch needs to contain, unlike Triplet Margin Loss
+which uses a `ClassSampler` to construct batches an equal amount of each class in the batch.
+Instead, we can construct batches using random sampling, a much simpler method which consequently takes less time to construct a batch.
+By default, runs created using ArcFace or Cosface loss will use random sampling, however you can specify which type of sampling method you would like to use like so:
+
+```diff
+run = finetuner.fit(
+    ...,
+    loss='ArcFaceLoss',
++   sampler='random'      # use random sampling
++   sampler='class'       # use class sampling
++   sampler='auto'        # infer sampling method based on the loss function (default)
+
+)
+```
+
+Since Triplet Margin Loss needs a class sampler to properly function, this `sampler` parameter is ignored when `loss=TripletMarginLoss`.
+
+### Using an Optimizer
+
+In order to keep track of the class centers across batches, these SphereFace loss functions require an additional optimizer during training.
+By default, the type of optimizer used will be the same as the one used for the model itself, but you can also choose a different optimizer for your loss function using the `loss_optimizer` parameter.
+The list of available optimizers are discussed in the [run job](../walkthrough/run-job.md) section.
+
+```diff
+run = finetuner.fit(
+    ...,
+    loss='ArcFaceLoss',
++   loss_optimizer='Adam',
++   loss_optimizer_options={'weight_decay': 0.01}
+)
+```
+
+Using these loss functions over the default Triplet Margin Loss can result in clearer divisions between the domains representing each class in the embedding space.
+As an example, the figure below shows the domains of the 10 classes of the FMNIST dataset projected onto 2D space after training with Triplet Loss, ArcFace Loss and CosFace Loss.
+
+![distributions-loss](../imgs/distributions-loss.png)
+
+Each color represents a different class. You can see how all of the loss functions are able to separate some of the classes from the others,
+but struggle to separate the green, blue, pink, purple and red classes,
+with Triplet Loss sperarating them the least, and ArcFace separating them the most.
+
+## Pooling layers
+
+Pooling layers are layers in a machine learning model that are used to reduce the dimensionality of data. Typically this is done in two ways, average pooling or max pooling.
+While a model may have many pooling layers within it, it is unwise to replace a pooling layer with another unless it is the last layer of the model.
+
```

Comment:
Maybe also mention that some models produce multiple embeddings, however, during metric learning we want to map the whole input into a single embedding representation